integration/e2emimir/client.go

```diff
@@ -1479,6 +1480,41 @@ func (c *Client) GetReceiversExperimental(ctx context.Context) ([]alertingmodels
 	return decoded, nil
 }
 
+func (c *Client) TestTemplatesExperimental(ctx context.Context, ttConf alertingNotify.TestTemplatesConfigBodyParams) (*alertingNotify.TestTemplatesResults, error) {
+	u := c.alertmanagerClient.URL("api/v1/grafana/templates/test", nil)
+
+	data, err := json.Marshal(ttConf)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling test templates config: %s", err)
+	}
+
+	req, err := http.NewRequest(http.MethodPost, u.String(), bytes.NewReader(data))
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %v", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, body, err := c.alertmanagerClient.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, ErrNotFound
+	}
+
+	if resp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("testing templates failed with status %d and error %v", resp.StatusCode, string(body))
```

Comment:
[Nit] Might as well be precise about the type:
```suggestion
		return nil, fmt.Errorf("testing templates failed with status %d and error %s", resp.StatusCode, string(body))
```